### PR TITLE
docs: update usage guide for changing models; add custom model support and reorganize sections

### PR DESCRIPTION
**Pull request by mrT23**
### **PR Type**
Documentation, Enhancement

___

### **Description**
- Enhanced `get_max_tokens` function to include `custom_model_max_tokens` from settings and added a detailed docstring.
- Removed detailed instructions for changing models from `additional_configurations.md`.
- Added a new `changing_a_model.md` file with detailed instructions for changing models, including sections for various providers like Azure, Hugging Face, and others.
- Updated the usage guide index and mkdocs navigation to include the new "Changing a Model" section.
- Added `custom_model_max_tokens` configuration option in `configuration.toml`.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>utils.py</strong><dd><code>Enhance `get_max_tokens` logic and add docstring</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/algo/utils.py

<li>Added detailed docstring for <code>get_max_tokens</code> function.<br> <li> Enhanced logic to include <code>custom_model_max_tokens</code> from settings.<br>

</details>

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/1066/files#diff-6b9df72d53c6f0d89fb142c210238a276c0782305e0024d16fbfcaf72c2e2b53">+13/-3</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>configuration.toml</strong><dd><code>Add custom model max tokens configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/settings/configuration.toml

- Added `custom_model_max_tokens` configuration option.

</details>

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/1066/files#diff-66cfda5143e484ee53ecf7aa0df7dca8ad0b181256f4b0675905db35bcbbae78">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>additional_configurations.md</strong><dd><code>Remove model changing instructions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/docs/usage-guide/additional_configurations.md

- Removed detailed instructions for changing models.

</details>

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/1066/files#diff-9290a3ad9a86690b31f0450b77acd37ef1914b41fabc8a08682d4da433a77f90">+0/-159</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>changing_a_model.md</strong><dd><code>Add detailed instructions for changing models</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/docs/usage-guide/changing_a_model.md

<li>Added detailed instructions for changing models.<br> <li> Included sections for various providers like Azure, Hugging Face, and <br>others.<br>

</details>

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/1066/files#diff-7904af6a8722d533e7de626761ae6c1d9554069ed02634c3b719d2e7f8b39f4d">+187/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>index.md</strong><dd><code>Update usage guide index with new sections</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/docs/usage-guide/index.md

<li>Added link to new "Changing a Model" section.<br> <li> Updated link to "PR-Agent Pro Models".<br>

</details>

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/1066/files#diff-dabd9ad2b5e65ed825e46f5ff0060386b51b62a0eedf4ca541aacee2b12af762">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>mkdocs.yml</strong><dd><code>Update mkdocs navigation with new sections</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/mkdocs.yml

<li>Added "Changing a Model" to navigation.<br> <li> Updated "PR-Agent Pro Models" link.<br>

</details>

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/1066/files#diff-7618337a88ea47f9b778a21f0e17d43eab2aba4880c6be2d95a7bf95e4b8b2df">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: pr_agent/algo/utils.py [hunk 1]
```diff
@@ -693,15 +693,25 @@ def get_user_labels(current_labels: List[str] = None):
...
+    (1) If the model is in './pr_agent/algo/__init__.py', use the value from there.
+    (2) else, the user needs to define explicitly 'config.custom_model_max_tokens'
+
+    For both cases, we further limit the number of tokens to 'config.max_model_tokens' if it is set.
+    This aims to improve the algorithmic quality, as the AI model degrades in performance when the input is too long.
+    """
     settings = get_settings()
     if model in MAX_TOKENS:
         max_tokens_model = MAX_TOKENS[model]
+    elif settings.config.custom_model_max_tokens > 0:
```
Comment (mrT23) [inline pr_agent/algo/utils.py:708]: this is an example line comment

**Comment by mrT23** [comment]
/review --pr_reviewer.extra_instructions="the generated output should be in Japanese"

**Comment by mrT23** [comment]
## Generating PR code suggestions


Work in progress ...<br>

**Comment by mrT23** [comment]
## Generating PR code suggestions


Work in progress ...<br>

**Comment by mrT23** [comment]
## Generating PR code suggestions


Work in progress ...<br>

**Review by hussam789 — APPROVED** [review]

**Pull request merged**